lampod/src/jsonrpc/offchain.rs

```diff
@@ -80,40 +79,69 @@ pub fn json_decode_invoice(ctx: &LampoDaemon, request: &json::Value) -> Result<j
             routes: Vec::new(),
             hints: Vec::new(),
             expiry_time: Some(invoice.expiry_time().as_millis() as u64),
-        }
-    } else if let Ok(offer) = ctx
+        };
+
+        return Ok(json::to_value(&bolt11_invoice)?);
+    }
+
+    if let Ok(offer) = ctx
         .offchain_manager()
         .decode::<ldk::offers::offer::Offer>(&request.invoice_str)
     {
-        // FIXME: simplify this chain;
-        let network = offer
+        let chains = offer
             .chains()
             .iter()
-            .map(|chain| Network::from_chain_hash(chain.clone()).unwrap().to_string())
-            .collect::<Vec<String>>()
+            .map(|chain| chain.to_string())
+            .collect::<Vec<String>>();
+
+        let network = offer
+            .chains()
             .first()
-            .unwrap_or(&Network::Bitcoin.to_string())
-            .clone();
-        InvoiceInfo {
-            issuer_id: offer.issuer().map(|id| id.to_string()),
-            amount_msat: offer.amount().map(|a| {
-                if let Amount::Bitcoin { amount_msats } = a {
-                    amount_msats.clone()
-                } else {
-                    unimplemented!()
-                }
-            }),
-            network: network.to_string(),
-            description: offer.description().map(|str| str.to_string()),
-            routes: Vec::new(),
-            hints: Vec::new(),
-            expiry_time: offer.absolute_expiry().map(|a| a.as_millis() as u64),
-        }
-    } else {
-        unreachable!()
-    };
+            .map(|hash| Network::from_chain_hash(*hash))
+            .unwrap_or(Some(Network::Bitcoin));
 
-    Ok(json::to_value(&invoice)?)
+        let paths = offer
+            .paths()
+            .to_vec()
+            .iter()
+            .map(|path| BlindedPath {
+                blinded_hops: path
+                    .blinded_hops
+                    .iter()
+                    .map(|node| node.blinded_node_id.to_string())
+                    .collect::<Vec<String>>(),
+                blinding_points: path.blinding_point.to_string(),
+            })
+            .collect::<Vec<BlindedPath>>();
+
+        //FIXME: Convert this to hex
+        let offer_id = format!("{:?}", offer.id().0);
+        let desc = offer
+            .description()
+            .map(|desc| desc.to_string())
+            .unwrap_or_else(|| "Description not provided".to_string());
+
+        let issuer_id = offer
+            .issuer()
+            .map(|issuer| issuer.to_string())
+            .unwrap_or_else(|| "No issuer id provided".to_string());
```

Comment:
When a type is None, do not force to be not none. Keep it `Option` otherwise the user of our API need to check the ID.

If there is no description, leave it None.

Reply:
Done d7ceae28b8acaf599a2e158ca7317844357e295b